ext/magic/ruby-magic.c

```diff
@@ -294,15 +294,19 @@ rb_mgc_get_paths(VALUE object)
 
 	MAGIC_CHECK_OPEN(object);
 
+	/* always prefer the value cached in @paths */
 	value = rb_ivar_get(object, id_at_paths);
-	if (!NIL_P(value) && !RARRAY_EMPTY_P(value) && !getenv("MAGIC"))
+	if (!NIL_P(value) && !RARRAY_EMPTY_P(value)) {
 		return value;
+	}
 
-	cstring = magic_getpath_wrapper();
-	value = magic_split(CSTR2RVAL(cstring), CSTR2RVAL(":"));
-	RB_GC_GUARD(value);
-
-	return value;
+	/* use either the library method or our gem-specific default, and save the result in @paths */
+	value = rb_funcall(rb_cMagic, rb_intern("default_paths"), 0);
+	if (getenv("MAGIC") || NIL_P(value)) {
+		cstring = magic_getpath_wrapper();
+		value = magic_split(CSTR2RVAL(cstring), CSTR2RVAL(":"));
+	}
+	return magic_set_paths(object, value);
```

Comment:
I don't know if it was coincidence, but my VM just stopped functioning after I ran this after installing the native package:

```
$ MAGIC=/tmp irb
irb(main):001:0> require 'magic'
=> true
irb(main):002:0> Magic.new
```

Reply:
Not seeing this problem after a reboot. 🤷